tasks/eva_2205/src/main/groovy/eva2205/eva2205_undo_merges_into_multimaps.groovy

```diff
@@ -0,0 +1,137 @@
+import uk.ac.ebi.ampt2d.commons.accession.core.models.EventType
+import uk.ac.ebi.eva.accession.core.GenericApplication
+import uk.ac.ebi.eva.accession.core.model.eva.ClusteredVariantInactiveEntity
+import uk.ac.ebi.eva.accession.core.model.eva.ClusteredVariantOperationEntity
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantInactiveEntity
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantOperationEntity
+import uk.ac.ebi.eva.accession.deprecate.Application
+import uk.ac.ebi.eva.groovy.commons.EVAObjectModelUtils
+import uk.ac.ebi.eva.groovy.commons.RetryableBatchingCursor
+
+import static org.springframework.data.mongodb.core.query.Query.query
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.*
+import static org.springframework.data.mongodb.core.query.Criteria.where
+import groovy.cli.picocli.CliBuilder
+
+def cli = new CliBuilder()
+cli.propertiesFile(args:1, "Properties file to use for remediation", required: true)
+cli.assembly(args:1, "Assembly to remediate", required: true)
+def options = cli.parse(args)
+if (!options) {
+    cli.usage()
+    System.exit(1)
+}
+
+def dbEnv = createFromSpringContext(options.propertiesFile, Application.class,
+        ["parameters.assemblyAccession": options.assembly])
+def dbsnpSSIDUpperBound = dbEnv.springApplicationContext.getBean("accessioningMonotonicInitSs", Long.class)
+def dbsnpRSIDUpperBound = dbEnv.springApplicationContext.getBean("accessioningMonotonicInitRs", Long.class)
+def logger = LoggerFactory.getLogger(GenericApplication.class)
+
+def undoMergeIntoMultiMapRS = {svoeOps ->
+    def (bulkSVEUpdateOps, bulkdbSnpSVEUpdateOps) =
+    [sveClass, dbsnpSveClass].collect { dbEnv.mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED, it) }
+    def (svoeInserts, dbSnpSvoeInserts, dbsnpCvoeInserts) = [new ArrayList<>(), new ArrayList<>(),
+                                                             new ArrayList<>()]
+    def (numSVEUpdates, numDbsnpSveUpdates, numDbsnpCvoeUpdates) = [0, 0, 0]
+
+    svoeOps.each {svoeOp ->
+        def ssHash = svoeOp.inactiveObjects[0].hashedMessage
+        Query queryToFindSSToUpdate = query(where("_id").is(ssHash).and("rs").exists(true))
+        def ssRecordsToUpdate = [sveClass, dbsnpSveClass].collect{
+            dbEnv.mongoTemplate.find(queryToFindSSToUpdate, it)}.flatten()
+        if (ssRecordsToUpdate.size() > 0) {
+            def ssAccession = ssRecordsToUpdate[0].accession
+            def isEvaSS = (ssAccession >= dbsnpSSIDUpperBound)
+            def (sveUpdateObj, svoeInsertObj) =  isEvaSS?
+                    [bulkSVEUpdateOps, svoeInserts] :[bulkdbSnpSVEUpdateOps, dbSnpSvoeInserts]
+            def oldRS = ssRecordsToUpdate[0].clusteredVariantAccession
+            // Since we are undoing a previous RS assignment to this SS,
+            // the new RS is the one from the Op recording that previous assignment
+            def newRS = svoeOp.inactiveObjects[0].clusteredVariantAccession
```

Comment:
Just to check my understanding, this whole method assumes that we're in the case where we have a non-map-weighted RS merged into a map-weighted RS, right?  And the recursion below takes care of traversing the "map-weighted to map-weighted" merges until we reach this case.